File: app/src/test/java/dev/bee/kanjianki/SettingsDeckLimitsPanelModelsTest.java
package dev.bee.kanjianki;

import dev.bee.kanjianki.core.RecordsSyncModels;
import dev.bee.kanjianki.core.SettingsTextCopy;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public final class SettingsDeckLimitsPanelModelsTest {
    @Test
    public void createUsesCurrentNewPerDayAndCopyContract() {
        RecordsSyncModels.Settings settings = RecordsSyncModels.Settings.kikuDefaults();
        SettingsDeckLimitsPanelModel model = SettingsDeckLimitsPanelModels.create(settings, text -> { });

        assertEquals(SettingsTextCopy.deckLimitsTitle(), model.getTitle());
        assertEquals(SettingsTextCopy.deckLimitsBody(), model.getBody());
        assertEquals(SettingsTextCopy.newCardsPerDayLabel(), model.getNewPerDayLabel());
        assertEquals(Integer.toString(settings.newPerDay), model.getInitialNewPerDayText());
        assertEquals(SettingsTextCopy.saveDeckLimitsLabel(), model.getSaveLabel());
        assertNotNull(model.getOnSave());
    }
}

File: app/src/test/java/dev/bee/kanjianki/SettingsDeckLimitsPanelModelsTest.kt
package dev.bee.kanjianki

import dev.bee.kanjianki.core.RecordsSyncModels
import dev.bee.kanjianki.core.SettingsTextCopy
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull
import org.junit.Test

class SettingsDeckLimitsPanelModelsTest {
    @Test
    fun createUsesCurrentNewPerDayAndCopyContract() {
        val settings = RecordsSyncModels.Settings.kikuDefaults()
        val model = SettingsDeckLimitsPanelModels.create(settings) { }

        assertEquals(SettingsTextCopy.deckLimitsTitle(), model.title)
        assertEquals(SettingsTextCopy.deckLimitsBody(), model.body)
        assertEquals(SettingsTextCopy.newCardsPerDayLabel(), model.newPerDayLabel)
        assertEquals(settings.newPerDay.toString(), model.initialNewPerDayText)
        assertEquals(SettingsTextCopy.saveDeckLimitsLabel(), model.saveLabel)
        assertNotNull(model.onSave)
    }
}
